massgov/mayflower
# DP-16109: Campaign key message overlay box positioning fix

Any PRs being created needs a changelog.txt file before being merged into dev. See: [Change Log Instructions](https://github.com/massgov/mayflower/blob/dev/docs/change-log-instructions.md)


## Description

When the background type option is set to "Solid color" in Drupal, a key message box is positioned where it's supposed to be with a background image because the component still render background image elements with its class settings. There was no validation with the background type options at Drupal side.

Since `key-message.twig` in Mayflower checks if `bgImage` is available to render or not.  As long as data is received, the background image related elements are rendered and classes to affect its display and the overlay box position are added.

Added a condition to check `bgImage` in `section` to add the class `ma__key-message--image-overlay` to affect the overlay box positioning.  Nothing changed to the background elements since if `bgImage` is null, nothing is rendered. 

## Related Issue / Ticket

- [JIRA issue](https://jira.mass.gov/browse/DP-16109)

## Steps to Test


1. Go to the following pages and confirm no display change with their current production version.
    - /patterns/05-pages-marketing-campaign/05-pages-marketing-campaign.html
    - /patterns/05-pages-marketing-campaign-as-solid/05-pages-marketing-campaign-as-solid.html

Production version:
- https://mayflower.digital.mass.gov/patterns/05-pages-marketing-campaign-as-solid/05-pages-marketing-campaign-as-solid.html
- https://mayflower.digital.mass.gov/patterns/05-pages-marketing-campaign-as-solid/05-pages-marketing-campaign-as-solid.html

## Screenshots
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.


## Additional Notes:

Anything else to add?

#### Impacted Areas in Application


* 

#### @TODO


*

#### Today I learned...
